### ISSUE TITLE
does this work in Node.js 18.9?

### ISSUE THREAD

**Issue by runvnc** [gptlabs/yield-stream#1]
I'm getting this: `TypeError: n.getReader is not a function
0|server  |     at o (file:///home/runvnc/siteaidev/node_modules/yield-stream/dist/lib.js:1:220)
0|server  |     at o.next (<anonymous>)
`

**Comment by ctjlewis** [comment]
Edit: Just realized this isn't the OpenAI streams lib.

I don't think I added Node support to this yet. I can probably do that really quick.

**Comment by runvnc** [comment]
It seems perfect otherwise. Appreciate your work. Do you have a donation link?

**Comment by ctjlewis** [comment]
No, but I appreciate it. 

Sorry about that by the way, I'll get a Node version out in an hour or so and ping you here.

**Comment by ctjlewis** [comment]
Also, I forgot to mention, NodeJS.Readable wasn't prioritized because WHATWG ReadableStream is available in Node 18+ - if switching to v18 is possible, it'll work out of the box.

This fix will be in tonight for you though.

**Comment by runvnc** [comment]
Hm. Yeah I was using 18.9 and tried upgrading to latest, 19.whatever, but still getting the error. If you are sure its supported, I think maybe there is some other more basic problem with what I am doing. Maybe OpenAI returned an error or something. I am looking into it.

**Comment by ctjlewis** [comment]
No, it's definitely a runtime thing, it's not your fault - can you send a screenshot and show me how you're using it? Is this in Next?

**Comment by runvnc** [comment]
No its plain old Node.js. Need like 5 minutes I just forked it.

**Comment by ctjlewis** [comment]
Yeah, it's tested for Node 18 currently: https://github.com/gptlabs/yield-stream/actions/runs/4237873575/jobs/7364315931 

This error is what you'd see if e.g. you ran this on Node 14, because there is no `ReadableStream` available. Sanity checking with `node -v` and stuff shows you're on 18? If you're trusting any other process to call `node` for you, it might be grabbing an old version at a different filepath.

**Comment by ctjlewis** [comment]
In any case I'll have a siloed `NodeJS.Readable`-only entry-point for you here in like another hour max.

**Comment by runvnc** [comment]
I am passed that error. Here is what I had wrong:
"Internally, streams are often manipulated using generators via `for await (const chunk of yieldStream(stream)) { ... }`. We recommend following this pattern if you find it intuitive."
That error went away when I just took the 'yieldStream' out. (I noticed it already had that in node.ts)
Now I have a Chat parse error but I think that will be easy to fix.  Sorry for the bother.
